mypy/checkexpr.py

```diff
@@ -682,30 +686,66 @@ def check_typeddict_call_with_dict(
             return AnyType(TypeOfAny.from_error)
 
     def check_typeddict_call_with_kwargs(
-        self, callee: TypedDictType, kwargs: Dict[str, Expression], context: Context
+        self,
+        callee: TypedDictType,
+        kwargs: List[Tuple[Optional[str], Expression]],
+        context: Context,
     ) -> Type:
-        if not (callee.required_keys <= set(kwargs.keys()) <= set(callee.items.keys())):
-            expected_keys = [
-                key
-                for key in callee.items.keys()
-                if key in callee.required_keys or key in kwargs.keys()
-            ]
-            actual_keys = kwargs.keys()
+        """Check TypedDict constructor of format Point(x=1, y=2)"""
+        # Infer types of item values and expand unpack operators
+        items: Dict[str, Tuple[Expression, Type]] = {}
+        sure_keys: List[str] = []
+        maybe_keys: List[str] = []  # Will contain non-required items of unpacked TypedDicts
+        for key, value_expr in kwargs:
+            if key is not None:
+                # Regular key and value
+                value_type = self.accept(value_expr, callee.items.get(key))
+                items[key] = (value_expr, value_type)
+                sure_keys.append(key)
+            else:
+                # Unpack operator (**) was used; unpack all items of the type of this expression into items list
+                value_type = self.accept(value_expr, callee)
+                proper_type = get_proper_type(value_type)
+                if isinstance(proper_type, TypedDictType):
```

Comment:
This will not work for unions of TypedDicts (or other more complex types). I'm open to suggestions to improve this